src/OpenTelemetry/Metrics/MetricStreamIdentity.cs

```diff
@@ -24,16 +24,19 @@ namespace OpenTelemetry.Metrics
         private static readonly StringArrayEqualityComparer StringArrayComparer = new StringArrayEqualityComparer();
         private readonly int hashCode;
 
-        public MetricStreamIdentity(Meter meter, string instrumentName, string unit, string description, Type instrumentType, string[] tagKeys, double[] histogramBucketBounds)
+        public MetricStreamIdentity(Instrument instrument, MetricStreamConfiguration metricStreamConfiguration)
         {
-            this.MeterName = meter.Name;
-            this.MeterVersion = meter.Version ?? string.Empty;
-            this.InstrumentName = instrumentName;
-            this.Unit = unit ?? string.Empty;
-            this.Description = description ?? string.Empty;
-            this.InstrumentType = instrumentType;
-
-            if (tagKeys != null && tagKeys.Length > 0)
+            this.MeterName = instrument.Meter.Name;
+            this.MeterVersion = instrument.Meter.Version ?? string.Empty;
+            this.InstrumentName = metricStreamConfiguration?.Name ?? instrument.Name;
+            this.Unit = instrument.Unit ?? string.Empty;
+            this.Description = metricStreamConfiguration?.Description ?? instrument.Description ?? string.Empty;
+            this.InstrumentType = instrument.GetType();
+            this.ViewId = metricStreamConfiguration?.ViewId;
+            this.MetricStreamName = $"{this.MeterName}.{this.MeterVersion}.{this.InstrumentName}";
+
+            var tagKeys = metricStreamConfiguration?.CopiedTagKeys;
+            if (tagKeys != null)
             {
                 this.TagKeys = new string[tagKeys.Length];
                 tagKeys.CopyTo(this.TagKeys, 0);
```

Comment:
Hmm. I guess this copying may no longer be required since we're making a copy of stuff further upstream. Likewise for histogram bounds.

Reply:
agree